metricq/source.py

```diff
@@ -89,31 +89,26 @@ async def task(self):
 
     """
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.metrics = dict()
-        self._chunk_size: int = 1
+    chunk_size: Optional[int] = cast(Optional[int], ChunkSize())
+    """Number of :term:`data points<Data Point>` collected into a chunk before being sent.
 
-    @property
-    def chunk_size(self) -> Optional[int]:
-        """Number of :term:`data points<Data Point>` collected into a chunk before being sent.
+    Initially, this value is set to :code:`1`, so any data point is sent immediately.
+    If set to :code:`None`, automatic chunking is disabled and data points must be sent off to the network manually using :meth:`flush`.
 
-        Initially, this value is set to :code:`1`, so any data point is sent immediately.
-        If set to :code:`None`, automatic chunking is disabled and data points must be sent off to the network manually using :meth:`flush`.
+    To reduce network and packet overhead, it may be advisable to send multiple data points at once.
+    Be aware that there is an `overhead-latency trade-off` to be made:
+    If your Source produces one data point every :math:`10` seconds, having a :code:`chunk_size` of :code:`10` means that it takes almost :math:`2` minutes (:math:`100` s) before a chunk is is sent.
+    If instead it produces :math:`1000` data points per second, network load can be reduced by setting a value of :code:`1000` without affecting latency too much.
 
-        To reduce network and packet overhead, it may be advisable to send multiple data points at once.
-        Be aware that there is an `overhead-latency trade-off` to be made:
-        If your Source produces one data point every :math:`10` seconds, having a :code:`chunk_size` of :code:`10` means that it takes almost :math:`2` minutes (:math:`100` s) before a chunk is is sent.
-        If instead it produces :math:`1000` data points per second, network load can be reduced by setting a value of :code:`1000` without affecting latency too much.
-        """
-        return None if self._chunk_size == 0 else self._chunk_size
+    Raises:
+        TypeError: if value set is neither :literal:`None` nor an integer
+        ValueError: if value set is not a positive integer
```

Comment:
Depending on the interpretation, this might or might not include `0` as valid input.
```suggestion
        ValueError: if value set is not a positive non-zero integer
```